lib/preview/previewUtils.js

```diff
@@ -0,0 +1,183 @@
+const { fetchDomains } = require('../../api/domains');
+const { getAccountId, isTrackingAllowed, getAccountConfig } = require('../config');
+const { platform, release } = require('os');
+const { trackUsage } = require('../../api/fileMapper');
+const unAuth = require('../../api/localDevAuth/unauthenticated');
+
+const VALID_PROXY_DOMAIN_SUFFIXES = ['localhost', 'hslocal.net'];
+const HS_PREVIEW_GATE = "cms:localHublPreviews";
+
+const getPortalDomains = async (portalId) => {
+  try {
+    const result = await fetchDomains(portalId);
+    return result;
+  } catch (error) {
+    console.log("There was a problem fetching domains for your portal. You may be missing a scope necessary for this feature.")
```

Comment:
Make sure to use `logger` rather than console, since it respects the logging level set by the user.